### DISCUSSION TITLE
[BETA] Userscripts v4.2.3 (macOS), v1.2.3 (iOS)

### DISCUSSION THREAD

**Discussion by quoid** [quoid/userscripts#300] [Announcements]
## Build 56/28

- [macOS] clarify acceptable blocklist values #283
- [macOS/iOS] fix userscript install checking bug when dom elements manipulated - #295
- [macOS/iOS] add GM.getTab and GM.saveTab - #296 (thank you @maggch97!)
- [macOS/iOS] further changed the way `@grant` interacts with `@inject-into` #265 
    - if `@inject-into` is not set, then `@inject-into` is set to `auto`
    - if `@inject-into` is `auto`
      - if `@grant`s are requested, then `@inject-into` is `content` 
      - if `@grant` is `none`, then `@inject-into` is `page`
        - if page content security policy blocks script execution, `@inject-into` becomes `content`
    - if `@inject-into` is `content`, all `@grant`s are provided
    - if `@inject-into` is `page`, no `@grant`s are provided (+ warning in the console)
    - thank you @scholtzm!

**Comment by scholtzm** [comment]
I signed up for macOS Beta, but would also test iOS Beta if possible. Do I need to signup again through the second Google form?

**Reply by quoid** [reply]
@scholtzm nope, I will put you on both - the beta builds are still in review but once that is done they will be available - Apple can take anywhere from a couple hours to a couple days to review testflight builds

**Comment by quoid** [comment]
## Build 57/29
- [macOS/iOS] update`@match` matching - #301

**Comment by quoid** [comment]
## Build 58/30
- [macOS/iOS] support for GM_info.script.icon - #302

**Comment by quoid** [comment]
## Build 59/31
- [macOS/iOS] - GM.xmlHttpRequest onloadstart bugfix - #304 (thank you @spd2last!)
- [macOS/iOS] - GM.getTab if tab is null, return empty object - #306 (thank you @maggch97!)

**Comment by quoid** [comment]
submitted for release today
